### ISSUE TITLE
Cannot import modules in `.nimble` during `nimble install`

### ISSUE THREAD

**Issue by madonuko** [nim-lang/nimble#1259]
`nimble install nimsutils` fails with the following error:
```
Downloading https://github.com/FyraLabs/nimsutils using git
   Warning: The package has no tagged releases, downloading HEAD instead.
  Verifying dependencies for nimsutils@0.1.0
 Installing nimsutils@0.1.0
       Tip: 3 messages have been suppressed, use --verbose to show them.
nimble.nim(399)          packageExists

    Error:  The package inside "/home/mado/.nimble/pkgs2/nimsutils-0.1.0-3d97d4dd06c39f7943ddb0b1dd28b05136077e63" is invalid.
  Details:  Could not read package info file in /home/mado/.nimble/pkgs2/nimsutils-0.1.0-3d97d4dd06c39f7943ddb0b1dd28b05136077e63/nimsutils.nimble;
        ...   Reading as ini file failed with: 
        ...     Invalid section: .
        ...   Evaluating as NimScript file failed with: 
        ...     /home/mado/.nimble/pkgs2/nimsutils-0.1.0-3d97d4dd06c39f7943ddb0b1dd28b05136077e63/nimsutils.nimble(1, 11) Error: cannot open file: src/nimsutils
        ... printPkgInfo() failed.
```
However, if you run `nimble` inside the [repository](https://github.com/FyraLabs/nimsutils) directly, it works absolutely fine.

Since I can't find any docs saying you aren't allowed to import things in the `.nimble` file, I feel like this should be a bug from `nimble install`.

**Comment by madonuko** [comment]
Have you tried running `nimble tasks`?

**Comment by xTrayambak** [comment]
![image](https://github.com/user-attachments/assets/59904313-ad7d-4560-8908-85d698c9c57c)
It does seem to parse it with no issues, but it cannot evaluate it.

**Comment by xTrayambak** [comment]
No, I mean "interpretation" when I say "evaluation". It cannot execute that.

**Comment by madonuko** [comment]
Sorry for my ignorance, but there a particular reason why `nimble install` can't import the file but in other situations you can?
The `xtask` thing clearly works (`nimble mytask` runs fine, `nimble tasks` can find it too, etc), I feel like the import system is broken particularly in `nimble install` only.

**Comment by daylinmorgan** [comment]
> here a particular reason why nimble install can't import the file but in other situations you can?

Yes, this is a combination of missing import paths and `nimble install` behavior. 
When nimble installs a package it only copies over the `srcDir` so the on-disk representation in the `.nimble/pkgs2` directory is like this:

```txt
nimsutils-0.1.0-e22ac7bf6b80f9650b3a2a0eb5871186c52e7a27
├── htmldocs
│   └── nimdoc.out.css
├── basedefs.nim
├── btrfmt.nim
├── common_imports.nim
├── exec.nim
├── nimblemeta.json
├── nimsutils.nim
├── nimsutils.nimble
├── time.nim
└── xtask.nim
```

Meaning that there is now no module at `./src/nimsutils.nim`.

A possible workaround is you doing some conditional import like this in your nimble file:

```nim
when not (compiles do: import src/nimsutils):
  import ./nimsutils
else:
  import src/nimsutils
```

**Comment by madonuko** [comment]
That unfortunately doesn't work (probably because Nim VM just fails on errors from `compiles do:`)

**Comment by daylinmorgan** [comment]
I would make sure to remove any versions of `nimsutils` from `~/.nimble` first. 

This works fine for me:

`nimble install https://github.com/daylinmorgan/nimsutils/`

What is the error? 
What version of `nim` and `nimble` are you using?

**Comment by madonuko** [comment]
Using latest devel (choosenim)

I have no idea why but somehow `nimble install https://github.com/daylinmorgan/nimsutils/` also works (probably some random nimble caching issues)

Maybe it's not a bad idea to document this behaviour somewhere?

**Comment by daylinmorgan** [comment]
> I have no idea why but somehow nimble install https://github.com/daylinmorgan/nimsutils/ also works (probably some random nimble caching issues) 
Ya probably,

TBF I think the original error message indicates the problem. 

But it may be good to add info to the `nimble-guide` about how packages are actually installed and how that relates to `scrDir`.
